lz4/stream/_stream.c

```diff
@@ -0,0 +1,1404 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of Steeve Morin nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software without
+ *    specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if defined(_WIN32) && defined(_MSC_VER)
+#define inline __inline
+#elif defined(__SUNPRO_C) || defined(__hpux) || defined(_AIX)
+#define inline
+#endif
+
+#include <py3c.h>
+#include <py3c/capsulethunk.h>
+#include <stdlib.h>
+#include <math.h>
+#include <lz4.h>
+#include <lz4hc.h>
+#include <stddef.h>
+#include <stdint.h>
+
+//#define DUMP_CONTEXT
+
+#if (LZ4_VERSION_MAJOR >= 1) && (LZ4_VERSION_MINOR >= 9) && (LZ4_VERSION_RELEASE >= 0)
+#define LZ4_VERSION_GE_1_9_0  1
+#pragma message "Building bindings against LZ4 " LZ4_VERSION_STRING ", using API>=1.9.0: yes"
+#else
+#define LZ4_VERSION_GE_1_9_0  0
+#pragma message "Building bindings against LZ4 " LZ4_VERSION_STRING ", using API>=1.9.0: no"
+#endif
+
+#ifndef Py_UNUSED /* This is already defined for Python 3.4 onwards */
+#ifdef __GNUC__
+#define Py_UNUSED(name) _unused_ ## name __attribute__((unused))
+#else
+#define Py_UNUSED(name) _unused_ ## name
+#endif
+#endif
+
+#if defined(_WIN32) && defined(_MSC_VER)
+#if _MSC_VER >= 1600
+#include <stdint.h>
+#else /* _MSC_VER >= 1600 */
+typedef signed char int8_t;
+typedef signed short int16_t;
+typedef signed int int32_t;
+typedef unsigned char uint8_t;
+typedef unsigned short uint16_t;
+typedef unsigned int uint32_t;
+#endif /* _MSC_VER >= 1600 */
+#if !defined(__CHAR_BIT__)
+#define __CHAR_BIT__ 8
+#endif /* __CHAR_BIT__ */
+#endif
+
+
+typedef enum
+{
+  COMPRESS,
+  DECOMPRESS,
+} lz4_direction_e;
+
+typedef enum
+{
+  DEFAULT,
+  FAST,
+  HIGH_COMPRESSION
+} compression_type;
+
+
+/* Forward declarations */
+static PyObject * LZ4StreamError;
+static PyTypeObject _LZ4StreamContext;
+
+#define DOUBLE_BUFFER_PAGE_COUNT          (2)
+
+#define DOUBLE_BUFFER_INDEX_MIN           (0)
+#define DOUBLE_BUFFER_INDEX_INVALID       (-1)
+
```

Comment:
Please reemove all this extra whitespace: none of the existing code does this whitespace-for-alignment stuff, which just adds noise to commits in the future.

As a general note: please chose to do one of these: either (1) Use indentation style consistent with `_frame.c` and `_block.c` (which is the standard GNU coding style, as implemented by default in emacs; or (2) comply with https://www.python.org/dev/peps/pep-0007/

In the long run, I'd like to have all C code compliant with PEP7. So, we can either use the stream stuff as the first occurence to that. Or, we need stream to be consistent with block and frame.

Reply:
done